pycytominer/operations/transform.py

```diff
@@ -67,58 +73,62 @@ def fit(self, X, y=None):
         self
             With computed weights attribute
         """
-        # Get the mean of the features (columns) and center if specified
-        self.mu = X.mean()
-        if self.center:
-            X = X - self.mu
+        # Get Numpy representation of the DataFrame
+        X = X.values
 
-        # Get the covariance matrix
-        C = (1 / X.shape[0]) * np.dot(X.transpose(), X)
-
-        if self.method in ["PCA", "ZCA"]:
-            # Get the eigenvalues and eigenvectors of the covariance matrix
-            s, U = eigh(C)
+        if self.method in ["PCA-cor", "ZCA-cor"]:
+            # The projection matrix for PCA-cor and ZCA-cor is the same as the
+            # projection matrix for PCA and ZCA, respectively, on the standardized
+            # data. So, we first standardize the data, then compute the projection
 
-            # Fix sign ambiguity of eigenvectors
-            U = pd.DataFrame(U * np.sign(np.diag(U)))
+            self.standard_scaler = StandardScaler().fit(X)
+            variances = self.standard_scaler.var_
+            if np.any(variances == 0):
+                raise ValueError(
+                    "Divide by zero error, make sure low variance columns are removed"
+                )
 
-            # Process the eigenvalues into a diagonal matrix and fix rounding errors
-            D = np.diag(1.0 / np.sqrt(s.clip(self.epsilon)))
+            X = self.standard_scaler.transform(X)
+        else:
+            if self.center:
+                self.mean_centerer = StandardScaler(with_mean=True, with_std=False).fit(
+                    X
+                )
+                X = self.mean_centerer.transform(X)
```

Comment:
ls there a reason to not set X as `X = StandardScaler(with_mean=True, with_std=False).fit_transform(X)`?

Reply:
Ah because I use `self.mean_centerer` later in `Spherize.transform`

```
if self.center:
    X = self.mean_centerer.transform(X)
```

Makes sense?